### ISSUE TITLE
Photo in horizontal orientation is saved stretched

### ISSUE THREAD

**Issue by eg-st-dev** [boxme/SquareCamera#8]

**Comment by philipgiuliani** [comment]
I can confirm this. I fixed it by cropping the image to a square before rotating it.

**Comment by bernardomesk** [comment]
@philipgiuliani can you show me how you did it?

**Comment by philipgiuliani** [comment]
@bernardomesk Im not in office anymore, i'll send you a codesnippet tomorrow.

**Comment by bernardomesk** [comment]
@philipgiuliani Thank you!

**Comment by philipgiuliani** [comment]
Thats how i changed the method in `EditSavePhotoFragment.java`

```java
public static final int SIZE = 960;

private void rotatePicture(int rotation, byte[] data, ImageView photoImageView) {
    Bitmap rawBitmap = BitmapFactory.decodeByteArray(data, 0, data.length);
    Bitmap bitmap = ThumbnailUtils.extractThumbnail(rawBitmap, SIZE , SIZE, ThumbnailUtils.OPTIONS_RECYCLE_INPUT);

    if (rotation != 0) {
        Matrix matrix = new Matrix();
        matrix.postRotate(rotation);

        bitmap = Bitmap.createBitmap(bitmap, 0, 0, bitmap.getWidth(), bitmap.getHeight(), matrix, false);
    }

    photoImageView.setImageBitmap(bitmap);
}
```

You could also calculate the SIZE, i just set it to 960 because i don't need bigger pictures. You could do the following to get the max square size:

```java
int size = Math.min(rawBitmap.getWidth(), rawBitmap.getHeight());
```

**Comment by bernardomesk** [comment]
@philipgiuliani This worked perfectly! One more thing. When you open the camera activity seems that the take picture button is duplicated. The layout with the button and the settings go up slightly. Did you notice that?

**Comment by philipgiuliani** [comment]
Yes its just an illusion... In fragment_camera.xml

```xml
<RelativeLayout
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    tools:context="com.desmond.squarecamera.CameraFragment">
```

also add

```
android:background="@android:color/black"
```

I hope thats what you meant. And that they go up slightly is because the application changes to fullscreen and the statusbar gets removed... No idea how to fix that.

**Comment by bernardomesk** [comment]
@philipgiuliani This is exactly what i wanted. It fixed another bug from the save image button that was left behind on camera fragment when you dismissed the image on save fragment. Thanks a lot and sorry for my bad english =)

**Comment by philipgiuliani** [comment]
No the selfie isn't streching for me... Maybe the camera has a to small resolution and you are upscaling it? Check bitmap.getWidth() and bitmap.getHeight().

**Comment by boxme** [comment]
Issue resolved in commit 90fa24a5f8a1cba99130fe22b413c2f237151381.
